packages/circuits/src/types/ProofGenerationError.ts

```diff
@@ -0,0 +1,9 @@
+export class ProofGenerationError extends Error {
+    constructor(message?: string) {
+        super(message)
+    }
+}
+
+export const InvalidProofInputError = new ProofGenerationError(
```

Comment:
I suggest initializing the error object at the moment the error occurs, and error message should describe the situation at that time. For example,

```js
function foo {
  if (somethingIsWrong) {
    throw new ProofGenerationError("something is wrong")
  }
}
```